README.md

```diff
@@ -112,6 +170,61 @@ kapt {
+
+To workaround this, the generated code attempts to match the bounds set by the sealed class, if it is able to.
+If it can't, due to variance or multiple bounds, it simply use a wildcard `*` instead.
+
+| Sealed Class | Sealed Enum |
+| --- | --- |
+| `sealed class Alpha`                              | `object AlphaSealedEnum : SealedEnum<Alpha>` |
+| `sealed class Beta<T>`                            | `object BetaSealedEnum : SealedEnum<Any?>` |
+| `sealed class Gamma<T : Omega>`                   | `object GammaSealedEnum : SealedEnum<Omega>` |
+| `sealed class Delta<T> where T : Omega, T : Psi`  | `object DeltaSealedEnum : SealedEnum<Delta<*>>` |
+| `sealed class Epsilon<out T : Omega>`             | `object EpsilonSealedEnum : SealedEnum<Epsilon<*>>` |
+| `sealed class Zeta<in T : Omega>`                 | `object ZetaSealedEnum : SealedEnum<Zeta<*>>` |
+| `sealed class Eta<out T>`                         | `object EtaSealedEnum : SealedEnum<Eta<*>>` |
+| `sealed class Theta<in T>`                        | `object ThetaSealedEnum : SealedEnum<Theta<*>>` |
+| `sealed class Iota<T : Omega, in U : Psi>`        | `object IotaSealedEnum : SealedEnum<Iota<Omega, *>>` |
+
+Similarly, the interfaces implemented by generated enum classes cannot directly have any variance or wildcards, so only simple generics will work.
+To avoid failing compilation, interfaces that are impossible to specify will be skipped.
+
+| Sealed Class | Enum Class |
+| --- | --- |
+| `sealed class Alpha`                                              | `enum class AlphaEnum` |
+| `sealed class Beta<T> : Sigma<T>`                                 | `enum class BetaEnum(sealedObject : Beta<Any?>) : Sigma<Any?> by sealedObject` |
+| `sealed class Gamma<T : Omega> : Sigma<T>`                        | `enum class GammaEnum(sealedObject : Gamma<Omega>) : Sigma<Omega> by sealedObject` |
+| `sealed class Delta<T> : Tau, Sigma<T> where T : Omega, T : Psi`  | `// Sigma skipped`<br/>`enum class DeltaEnum(sealedObject: Delta<*>) : Tau` |
+| `sealed class Epsilon<out T : Omega> : Sigma<T>`                  | `// Sigma skipped`<br/>`enum class EpsilonEnum` |
+| `sealed class Zeta<in T : Omega> : Sigma<T>`                      | `// Sigma skipped`<br/>`enum class ZetaEnum` |
+| `sealed class Eta<out T> : Tau, Sigma<T>`                         | `// Sigma skipped`<br/>`enum class EtaEnum(sealedObject: Eta<*>) : Tau` |
+| `sealed class Theta<in T> : Tau, Sigma<T>`                        | `// Sigma skipped`<br/>`enum class ThetaEnum(sealedObject: Theta<*>) : Tau` |
+| `sealed class Iota<T : Omega, in U : Psi> : Sigma<T>, Upsilon<U>` | `// Upsilon skipped`<br/>`enum class IotaEnum(sealedObject: Iota<Omega, *>): Sigma<Omega>` |
+| `sealed class Kappa<out T> : Sigma<Upsilon<T>>`                   | `enum class KappaEnum(sealedObject: Kappa<*>) : Sigma<Upsilon<*>>` |
```

Comment:
My main objective here was to avoid failing compilation, since it's a pretty big task to traverse all of the super classes and accumulate all of the interfaces. I felt like there's a pretty good chance that the interface you'd want from the `enum` might be completely separate from an interface that isn't valid for the enum. The other advantage in this case is that an interface that won't be valid for the enum won't be valid for _any_ enum, so hopefully it'd make it a bit clearer why an interface isn't going on the enum.

That being said, I agree that quietly sweeping the issues under the bed isn't the most helpful. I think I like both of those: a command line warning upon annotation processing, warning that an interface won't be implemented by the enum class, and a comment within the generated code would be sweet for debugging purposes. Both of those are silent enough that if it's intentional it can be ignored, and if not it will help elevate it to be fixed.

Reply:
I opened #12 to address this issue.